crates/bevy_render/src/settings.rs

```diff
@@ -93,6 +96,40 @@ impl Default for WgpuSettings {
     }
 }
 
+/// An enum describing how the renderer will initialize resources. This is used when creating the [`RenderPlugin`](crate::RenderPlugin).
+pub enum RenderSettings {
+    /// Allows renderer resource initialization to happen outside of the rendering plugin.
+    /// Because [`Instance`] can't be cloned, it needs to be wrapped in a Mutex so its value can be taken.
+    Manual(
+        RenderDevice,
+        RenderQueue,
+        RenderAdapterInfo,
+        RenderAdapter,
+        Mutex<Instance>,
```

Comment:
the mutex here is to allow interior mutability, which breaks the assumption that setting up a plugin doesn't mutate it

Reply:
from what I could tell, all methods on `wgpu::Instance` are on a shared ref, and the type is `Send` and `Sync`. Is there a reason that cloning a `Arc<wgpu::Instance>` cannot be used instead taking from a `Mutex<Instance>`?

Reply:
@mockersf my most recent commit should have fixed this. i implemented what @TheButlah recommended